media/system/js/bootstrap-init.js

```diff
@@ -1,142 +1,146 @@
-;(function() {
-	"use strict";
+((document, getOptions, $) => {
+  "use strict";
 
-	jQuery(document).ready(function($) {
+  if (typeof getOptions !== 'function') {
+    throw new Error('Joomla API is not properly initialised');
+  }
 
-		// Initialize some variables
-		var accordion = Joomla.getOptions('bootstrap.accordion'),
-		    alert = Joomla.getOptions('bootstrap.alert'),
-		    button = Joomla.getOptions('bootstrap.button'),
-		    carousel = Joomla.getOptions('bootstrap.carousel'),
-		    dropdown = Joomla.getOptions('bootstrap.dropdown'),
-		    modal = $('.joomla-modal'),
-		    popover = Joomla.getOptions('bootstrap.popover'),
-		    scrollspy = Joomla.getOptions('bootstrap.scrollspy'),
-		    tabs = Joomla.getOptions('bootstrap.tabs'),
-		    tooltip = Joomla.getOptions('bootstrap.tooltip');
+  $(document).ready(($) => {
 
-		/** Accordion **/
-		if (accordion) {
-			$.each(accordion, function(index, value) {
-				$('#' + index).collapse(
-					{
-						parent:value.parent,
-						toggle:value.toggle
-					}
-				).on("show", new Function(value.onShow)())
-					.on("shown", new Function(value.onShown)())
-					.on("hideme", new Function(value.onHide)())
-					.on("hidden", new Function(value.onHidden)());
-			});
-		}
+    // Initialize some variables
+    var accordion = getOptions('bootstrap.accordion'),
+      alert = getOptions('bootstrap.alert'),
+      button = getOptions('bootstrap.button'),
+      carousel = getOptions('bootstrap.carousel'),
+      dropdown = getOptions('bootstrap.dropdown'),
+      modal = $('.joomla-modal'),
+      popover = getOptions('bootstrap.popover'),
+      scrollspy = getOptions('bootstrap.scrollspy'),
+      tabs = getOptions('bootstrap.tabs'),
+      tooltip = getOptions('bootstrap.tooltip');
 
-		/** Alert **/
-		if (alert) {
-			$.each(alert, function(index, value) {
-				$('#' + index).alert();
-			});
-		}
+    /** Accordion **/
+    if (accordion) {
+      $.each(accordion, (index, value) => {
+        $('#' + index).collapse(
+          {
+            parent: value.parent,
+            toggle: value.toggle
+          }
+        ).on("show", new Function(value.onShow)())
+          .on("shown", new Function(value.onShown)())
+          .on("hideme", new Function(value.onHide)())
+          .on("hidden", new Function(value.onHidden)());
+      });
+    }
 
-		/** Button **/
-		if (button) {
-			$.each(button, function(index, value) {
-				$('#' + index).button();
-			});
-		}
+    /** Alert **/
+    if (alert) {
+      $.each(alert, (index, value) => {
```

Comment:
I know this is BS, but the `$.each` can be made vanilla

Reply:
this file will be moved to legacy and will never be used in J4 (at least for the core and hopefully by 3rd PD as BS components are not really accessible)